cfme/ansible_tower/jobs.py

```diff
@@ -51,14 +61,79 @@ def is_displayed(self):
         )
 
 
+class AnsibleTowerJobsDetailsView(TowerJobsView):
+    @View.nested
+    class entities(View):  # noqa
+        """ Represents details page when it's switched to Summary/Table view """
+        properties = SummaryTable(title="Properties")
+        relationships = SummaryTable(title="Relationships")
+        smart_management = SummaryTable(title="Smart Management")
+
+    @property
+    def is_displayed(self):
+        """Is this view being displayed?"""
+        return self.title.text == '{} (Summary)'.format(self.context['object'].template_name)
+
+
 @attr.s
-class TowerJobs(BaseEntity):
-    pass
+class TowerJob(BaseEntity):
+    template_name = attr.ib()
+
+    @property
+    def status(self):
+        view = navigate_to(self, 'Details')
+        return view.entities.properties.get_text_of("Status")
+
+    def delete(self):
+        view = navigate_to(self, 'Details')
+        view.toolbar.configuration.item_select('Remove this Job', handle_alert=True)
+        view.flash.assert_no_error()
+
+    def is_job_successful(self):
+        return True if self.status == "successful" else False
+
+    def wait_for_completion(self, num_sec=1200, delay=10):
+        def last_status():
+            logger.info("Last status message in UI: '{}'".format(self.status))
+
+        wait_for(self.is_job_successful, num_sec=num_sec, delay=delay, fail_func=last_status,
+                 message="Job finished")
 
 
 @attr.s
 class TowerJobsCollection(BaseCollection):
-    ENTITY = TowerJobs
+    ENTITY = TowerJob
+
+    def all(self):
+        view = navigate_to(self, 'All')
+        return [
+            self.instantiate(template_name=e.data["template_name"])
+            for e in view.entities.get_all(surf_pages=True)
+        ]
+
+    def status(self, template_name):
+        """Get status of a specific job from the All page."""
+        view = navigate_to(self, 'All')
+        for e in view.entities.get_all(surf_pages=True):
+            if e.data["template_name"] == template_name:
+                return e.data["status"]
+        raise NoSuchElementException("No job named {}".format(template_name))
+
+    def delete_all(self):
+        view = navigate_to(self, 'All')
+        view.paginator.check_all()
```

Comment:
I believe this is only deleting all the entities on the current page.

Mixed thoughts on a convenience method like this being added to a collection.

A far more useful implementation would take a list of entities to select and delete, with an 'all' kwarg to handle selection via paginator.  Combined with standardized definition of the configuration string used to delete the entities, this could be implemented in BaseCollection.

I don't see any uses of this method in your current tests, were you adding it for future use? Perhaps a large-scope fixture that could cleanup everything after a tower test module runs?